thcrap_proton

```diff
@@ -86,5 +88,18 @@ echo "--------------------GENERATED COMMAND--------------------"
 echo "$COMMAND"
 echo "---------------------------------------------------------"
 
-sh -c "$COMMAND"
+sh -c "$COMMAND" &
+
+# Get PID while ensuring only the actual Touhou executable is chosen
+while [ -z "$TH_PID" ]; do # Loop until a matching process is found
+    TH_PID=$(pgrep -a "$TH_REGEX" | grep -iE "$TH_REGEX" | awk '{print $1}')
```

Comment:
Is the second ```grep``` really needed? Also, if we remove the ```-a``` flag, we can avoid the ```awk```.
```suggestion
    TH_PID=$(pgrep "$TH_REGEX")
```

Reply:
Huh, if I didn't have the `grep` in there it also grabbed other related processes like the `thcrap_loader` itself. Just tested it with your suggestions and it seems to work just fine. Must've been the weird regex I had before, if I had to guess.